eras/shelley/impl/src/Cardano/Ledger/Shelley/API/Wallet.hs

```diff
@@ -515,26 +521,22 @@ evaluateTransactionFee pp tx numKeyWits = getMinFeeTx pp tx'
 evaluateTransactionBalance ::
   ( EraUTxO era,
     ShelleyEraTxBody era,
+    DepositInfo era ~ DPState (EraCrypto era),
     HasField "_poolDeposit" (PParams era) Coin,
     HasField "_keyDeposit" (PParams era) Coin
   ) =>
-  -- | The current protocol parameters.
+  -- | Current protocol parameters
   PParams era ->
+  -- | Where the deposit info is stored
+  DepositInfo era ->
   -- | The UTxO relevant to the transaction.
   UTxO era ->
-  -- | A predicate that a stake pool ID is new (i.e. unregistered).
-  -- Typically this will be:
-  --
-  -- @
-  --   (`Map.notMember` stakepools)
-  -- @
-  (KeyHash 'StakePool (EraCrypto era) -> Bool) ->
```

Comment:
This was on optimization that we intentionally added for the wallet team, namely to not have to pass the whole map. I forget the details, but we need to make sure that removing this is okay. It looks like `keyCertsRefunds` needs the tracked deposits, so perhaps there is nothing better we can do. cc @HeinrichApfelmus

Reply:
The problem is you can't compute this without knowing which Keys and Pools are already registered. We could make a version that took two maps as input, and then use them?
When we know what they need we can make some adjustments.